src/ophyd_async/core/signal.py

```diff
@@ -66,36 +74,49 @@ def __init__(
 
         super().__init__(name)
 
+    def children(self) -> Iterator[Tuple[str, "Device"]]:
+        yield from ()
+
     async def connect(
         self, mock=False, timeout=DEFAULT_TIMEOUT, force_reconnect: bool = False
     ):
         """Connect self with a timeout
 
-        Args:
-            mock (bool, optional): _description_. Defaults to False.
-            timeout (_type_, optional): _description_. Defaults to DEFAULT_TIMEOUT.
-            force_reconnect (bool, optional): _description_. Defaults to False.
+        Parameters
+        ----------
+        mock:  (bool, optional): uses the `MockSignalBackend. Defaults to False.
+        timeout:  (_type_, optional): Time to wait before failing with a TimeoutError. Defaults to DEFAULT_TIMEOUT.
+        force_reconnect:  (bool, optional): Always force reconnection. Defaults to False.
+
+        Notes
+        -----
+
+
         """
         can_use_previous_connection: bool = (
-            self._connect_task and not (self._connect_task.done() and self._connect_task.exception())
+            self._connect_task
+            and not (self._connect_task.done() and self._connect_task.exception())
+            and self._connect_mock_arg == mock
         )
         if force_reconnect or not can_use_previous_connection:
+            self._backend = (
+                MockSignalBackend(initial_backend=self._initial_backend)
+                if mock
+                else self._initial_backend
+            )
+
+            # Kick off a connection
             self.log.debug(f"Connecting to {self.source}")
-            self._connect_task = asyncio.create_task(wait_for_connection(self._backend.connect(timeout=timeout)))
+            self._connect_task = asyncio.create_task(
+                wait_for_connection(self._backend.connect(timeout=timeout))
```

Comment:
This is a single coro, so no need for `wait_for_connection`:
```suggestion
                self._backend.connect(timeout=timeout)
```

Reply:
applied this